### DISCUSSION TITLE
Any got luck flashing bootloader with RPi and OpenOCD?

### DISCUSSION THREAD

**Discussion by victpork** [makerdiary/nrf52840-mdk-usb-dongle#92] [Q&A] [answered]
Want to flash the bootloader to the latest version (0.2.13-42-g82464f9-dirty for mine). But haven't got any DAP probe at hand and I was thinking of using my RPi3 as alternative. However can't get it working as it shows "Error connecting DP: cannot read IDR" when starting a connection.

**Comment by blsmit5728** [comment]
I actually just did this with a PicoProbe, or at least the PicoProbe firmware on a Pi Pico board. 
Firmware for Pico: https://github.com/rgrr/yapicoprobe/
Wiring: 
Pico Pin 4->CLK
Pico Pin 5-> DIO 
Pico Pin 9 -> RST 
Pico Pin 36 -> 3V3
Pico Pin 38 -> GND
Using [XPACKOpenOCD](https://xpack.github.io/openocd/) for there I didn't write down the line I used to run openocd.... but I found it googling around.

**Comment by blsmit5728** [comment, marked as answer]
Sites that were helpful: 
[https://community.element14.com/products/roadtest/b/blog/posts/tutorial-02-part-2-debugging-nrf52840-using-raspberry-pi-arduino-nano-33-ble-sense-roadtest](Element14) 
[https://asmw.de/posts/bootloader-fix-for-makerdiary-nrf52840-mdk-usb-dongle/](this is what I based my command off of)
Using the two above I was able to get the flashing to work.

**Comment by idreamsi** [comment]
Has anyone done this without using DAP probe?
I also only have access to RPI Pico and Raspberry Pi.
@victpork Did you succeed?

**Reply by idreamsi** [reply]
Finally, I used the following method and quickly came back.
https://github.com/idreamsi/bootloader-fix-for-makerdiary-nrf52840-mdk-usb-dongle/

**Comment by victpork** [comment]
Hi,

Yup, after following @blsmit5728 instructions I flashed yapicoproble onto a pico and flashed the bootloader onto the mdk dongle.
